aptos-move/framework/aptos-framework/sources/overloadable_fungible_asset.move

```diff
@@ -0,0 +1,169 @@
+/// This defines the fungible asset module that can issue fungible asset of any `Metadata` object. The
+/// metadata object can be any object that equipped with `Metadata` resource.
+///
+/// The overloadable_fungible_asset wraps the existing fungible_asset module and adds the ability for token issuer
+/// to customize the logic for withdraw and deposit operations. For example:
+///
+/// - Deflation token: a fixed percentage of token will be destructed upon transfer.
+/// - Transfer allowlist: token can only be transfered to addresses in the allow list.
+/// - Predicated transfer: transfer can only happen when some certain predicate has been met.
+/// - Loyalty token: a fixed loyalty will be paid to a designated address when a fungible asset transfer happens
+///
+/// See AIP-73 for further discussion
+///
+module aptos_framework::overloadable_fungible_asset {
+    use aptos_framework::fungible_asset::{Self, FungibleAsset, TransferRef};
+    use aptos_framework::function_info::{Self, FunctionInfo};
+    use aptos_framework::object::{Self, ConstructorRef, Object};
+
+    use std::error;
+    use std::string;
+    use std::signer;
+
+    /// Provided withdraw function type doesn't meet the signature requirement.
+    const EWITHDRAW_FUNCTION_SIGNATURE_MISMATCH: u64 = 1;
+    /// Provided deposit function type doesn't meet the signature requirement.
+    const EDEPOSIT_FUNCTION_SIGNATURE_MISMATCH: u64 = 2;
+    /// Calling overloadable api on non-overloadable fungible asset store.
+    const EFUNCTION_STORE_NOT_FOUND: u64 = 3;
+    /// Recipient is not getting the guaranteed value;
+    const EAMOUNT_MISMATCH: u64 = 4;
+    /// Trying to register overload functions to fungible asset that has already been initialized with custom transfer function.
+    const EALREADY_REGISTERED: u64 = 5;
+    /// Fungibility is only available for non-deletable objects.
+    const EOBJECT_IS_DELETABLE: u64 = 18;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct OverloadFunctionStore has key {
+		withdraw_function: FunctionInfo,
```

Comment:
maybe these should be optional and we need balance and something else, @movekevin